rpc-interface/src/types.rs

```diff
@@ -79,6 +79,34 @@ impl Display for ValidityStartHeight {
     }
 }
 
+impl<'de> Deserialize<'de> for ValidityStartHeight {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        #[serde(untagged)]
+        enum StrOrU32 {
+            Number(u32),
+            String(String),
+        }
+
+        match StrOrU32::deserialize(deserializer)? {
+            StrOrU32::Number(height) => Ok(Self::Absolute(height)),
+            StrOrU32::String(height) => height
+                .trim()
+                .strip_prefix('+')
+                .map_or_else(
+                    || height.parse::<u32>().map(Self::Absolute), // In the case string is not prefixed by '+'
+                    |stripped| stripped.parse::<u32>().map(Self::Relative), // In the case string is prefixed by '+'
+                )
+                .map_err(|_| {
+                    Error::custom("provided ValidityStartHeight cannot be converted into a `u32`")
+                }),
+        }
```

Comment:
Instead of re-implementing the string parsing, you could just use the existing `FromStr` implementation below? Or otherwise remove that I guess?

Reply:
This could already work?
```suggestion
            StrOrU32::String(height) => Self::from_str(height),
        }
```